src/replicator.rs

```diff
@@ -613,8 +613,11 @@ impl Replicator {
                 std::mem::transmute(&callback),
             );
 
-            CBLReplicator_Stop(self.get_ref());
-            let success = receiver.recv_timeout(Duration::from_secs(10)).is_ok();
+            let mut success = true;
+            if self.status().activity != ReplicatorActivityLevel::Stopped {
+                CBLReplicator_Stop(self.get_ref());
+                success = receiver.recv_timeout(Duration::from_secs(10)).is_ok();
```

Comment:
Maybe we could have a public setter of this value to avoid potentially having to update the library

Reply:
On which value ?

Reply:
`Durration::from_secs(10)` I think

Reply:
Timeout added as parameter